app/code/Magento/ImportExport/Model/ResourceModel/Import/Data/Iterator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\ImportExport\Model\ResourceModel\Import\Data;
+
+/**
+ * Import data iterator
+ */
+class Iterator extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb
+{
+    /**
+     * Ids of rows in table
+     * @var array
+     */
+    protected $rowsIds = [];
+
+    /**
+     * Count of all rows in table
+     * @var int
+     */
+    protected $rowsCount = 0;
+
+    /**
+     * Current iterator index
+     * @var int
+     */
+    protected $currentIndex = 0;
+
+    /**
+     * Iterator constructor.
+     * @param \Magento\Framework\Model\ResourceModel\Db\Context $context
+     * @param null $connectionName
+     */
+    public function __construct(
+        \Magento\Framework\Model\ResourceModel\Db\Context $context,
+        $connectionName = null
+    ) {
+        parent::__construct($context, $connectionName);
+
+        $this->rowsIds = $this->getRowsIds();
```

Comment:
According to [Magento Technical Guidelines](http://devdocs.magento.com/guides/v2.1/coding-standards/technical-guidelines/technical-guidelines.html#class-design) it is strongly discouraged to perform any operations in the constructor of the class with the exception of Dependency Assignment and Dependency Validation operations.